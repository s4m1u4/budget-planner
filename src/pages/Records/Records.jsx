import React, { useCallback, useEffect, useState } from "react";
import { calculateRecords } from "../../helpers";
import { CircularProgress, Container } from "@mui/material";
import { FiltersSection } from "./components/FiltersSection";
import { RecordsSection } from "./components/RecordsSection";

import { Grid, ProgressWrapper, RecordsTitle, Wrapper } from "./Records.styles";

export const Records = ({
  limit,
  histories,
  getHistories,
  getFilteredHistories,
  deleteHistory,
  deleteAllHistories,
  categories,
  getCategories,
}) => {
  const records = calculateRecords(histories, categories);

  const [page, setPage] = useState(1);
  const [isLoading, setIsLoading] = useState(false);
  const [filters, setFilters] = useState({ type: "", category: "" });

  const fetchData = useCallback(async () => {
    setIsLoading(true);
    await getCategories();
    await getHistories(1, null);
    setIsLoading(false);
  }, [getCategories, getHistories]);

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return (
    <Container>
      {isLoading ? (
        <ProgressWrapper>
          <CircularProgress />
        </ProgressWrapper>
      ) : (
        <Wrapper>
          <RecordsTitle>Records page</RecordsTitle>
          <Grid>
            <FiltersSection
              page={page}
              setPage={setPage}
              filters={filters}
              setFilters={setFilters}
              categories={categories}
              getHistories={getHistories}
              getFilteredHistories={getFilteredHistories}
            />
            <RecordsSection
              page={page}
              setPage={setPage}
              limit={limit}
              filters={filters}
              records={records}
              getFilteredHistories={getFilteredHistories}
              deleteHistory={deleteHistory}
              deleteAllHistories={deleteAllHistories}
            />
          </Grid>
        </Wrapper>
      )}
    </Container>
  );
};
