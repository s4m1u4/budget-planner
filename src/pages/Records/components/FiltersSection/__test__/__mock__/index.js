export { CATEGORIES, HISTORIES } from "./mock";
