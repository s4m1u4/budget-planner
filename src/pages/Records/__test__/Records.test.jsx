import { Records } from "../Records";
import { act, render, screen } from "@testing-library/react";
import { CATEGORIES, HISTORIES } from "./__mock__";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";

const renderComponent = () => {
  return render(
    <DndProvider backend={HTML5Backend}>
      <Records
        limit={10000}
        categories={CATEGORIES}
        histories={HISTORIES}
        getHistories={getHistories}
        deleteHistory={deleteHistory}
        getCategories={getCategories}
        deleteAllHistories={deleteAllHistories}
        getFilteredHistories={getFilteredHistories}
      />
    </DndProvider>
  );
};

const renderComponentAsync = async () => {
  await act(async () => {
    await render(
      <DndProvider backend={HTML5Backend}>
        <Records
          limit={10000}
          categories={CATEGORIES}
          histories={HISTORIES}
          getHistories={getHistories}
          deleteHistory={deleteHistory}
          getCategories={getCategories}
          deleteAllHistories={deleteAllHistories}
          getFilteredHistories={getFilteredHistories}
        />
      </DndProvider>
    );
  });
};

const getHistories = jest.fn();
const deleteHistory = jest.fn();
const getCategories = jest.fn();
const deleteAllHistories = jest.fn();
const getFilteredHistories = jest.fn();

describe("Records page", () => {
  it("snapshot matching", async () => {
    await act(async () => {
      const { baseElement } = await renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe("should render", () => {
    beforeEach(async () => {
      await renderComponentAsync();
    });

    it("a title", () => {
      expect(screen.getByText(/records page/i)).toBeInTheDocument();
    });
  });
});
