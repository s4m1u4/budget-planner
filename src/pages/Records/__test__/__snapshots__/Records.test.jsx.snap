// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Records page snapshot matching 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-o6fnr7"
      >
        <h1
          class="css-15poin4"
        >
          Records page
        </h1>
        <div
          class="css-1go8ryl"
        >
          <div
            class="css-ub8u8e"
          >
            <div
              class="css-zeiouw"
            >
              <div
                class="css-1caytb6"
              >
                <h3
                  class="css-1xqtn9i"
                >
                  Filters
                </h3>
              </div>
              <div
                class="css-1gk4ta0"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-18gebku-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-8wl1e5-MuiButtonBase-root-MuiAccordionSummary-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        Type
                      </p>
                    </div>
                    <div
                      class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ExpandMoreIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        >
                          <div
                            class="MuiAccordionDetails-root css-1kh8flv-MuiAccordionDetails-root"
                          >
                            <fieldset
                              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                            >
                              <div
                                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                                role="radiogroup"
                              >
                                <label
                                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                                >
                                  <span
                                    class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                                  >
                                    <input
                                      class="PrivateSwitchBase-input css-1m9pwf3"
                                      type="radio"
                                      value="income"
                                    />
                                    <span
                                      class="css-hyxlzm"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                        data-testid="RadioButtonUncheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                        />
                                      </svg>
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                        data-testid="RadioButtonCheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                                  >
                                    Income
                                  </span>
                                </label>
                                <label
                                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                                >
                                  <span
                                    class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                                  >
                                    <input
                                      class="PrivateSwitchBase-input css-1m9pwf3"
                                      type="radio"
                                      value="expense"
                                    />
                                    <span
                                      class="css-hyxlzm"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                        data-testid="RadioButtonUncheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                        />
                                      </svg>
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                        data-testid="RadioButtonCheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                                  >
                                    Expense
                                  </span>
                                </label>
                              </div>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-18gebku-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-8wl1e5-MuiButtonBase-root-MuiAccordionSummary-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        Category
                      </p>
                    </div>
                    <div
                      class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ExpandMoreIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        >
                          <div
                            class="MuiAccordionDetails-root css-1kh8flv-MuiAccordionDetails-root"
                          >
                            <fieldset
                              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                            >
                              <div
                                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                                role="radiogroup"
                              >
                                <label
                                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                                >
                                  <span
                                    class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                                  >
                                    <input
                                      class="PrivateSwitchBase-input css-1m9pwf3"
                                      type="radio"
                                      value="084a3558-6b14-4e38-9d24-47e912021384"
                                    />
                                    <span
                                      class="css-hyxlzm"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                        data-testid="RadioButtonUncheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                        />
                                      </svg>
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                        data-testid="RadioButtonCheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                                  >
                                    Work
                                  </span>
                                </label>
                                <label
                                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                                >
                                  <span
                                    class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                                  >
                                    <input
                                      class="PrivateSwitchBase-input css-1m9pwf3"
                                      type="radio"
                                      value="0ce314b9-a547-4f57-9d64-e466777ba8e9"
                                    />
                                    <span
                                      class="css-hyxlzm"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                        data-testid="RadioButtonUncheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                        />
                                      </svg>
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                        data-testid="RadioButtonCheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                                  >
                                    Travel
                                  </span>
                                </label>
                                <label
                                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                                >
                                  <span
                                    class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                                  >
                                    <input
                                      class="PrivateSwitchBase-input css-1m9pwf3"
                                      type="radio"
                                      value="ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2"
                                    />
                                    <span
                                      class="css-hyxlzm"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                        data-testid="RadioButtonUncheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                        />
                                      </svg>
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                        data-testid="RadioButtonCheckedIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                                  >
                                    Education
                                  </span>
                                </label>
                              </div>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="css-p2xvpm"
                >
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-18gfzj-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    Search
                  </button>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-rk05t6-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
            <div
              class="css-zeiouw"
            >
              <div
                class="css-1caytb6"
              >
                <h3
                  class="css-1xqtn9i"
                >
                  Overview
                </h3>
              </div>
              <div
                class="css-1gk4ta0"
                style="background-color: inherit;"
              >
                <p>
                  Drag and drop records here🤚
                </p>
              </div>
            </div>
          </div>
          <div
            class="css-zeiouw"
          >
            <div
              class="css-1caytb6"
            >
              <h3
                class="css-1xqtn9i"
              >
                Records
              </h3>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1ht6x5l-MuiButtonBase-root-MuiIconButton-root"
                data-modal="data-delete"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteForeverRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="css-1gk4ta0"
            >
              <div
                class="css-u5hrl4"
              >
                <div
                  class="css-1mavqmw"
                  draggable="true"
                  style="background-color: inherit;"
                >
                  <p>
                    income
                     from 
                    Work
                  </p>
                  <div
                    class="css-1ub4dmn-MuiStack-root"
                  >
                    <span
                      class="css-81iyw9"
                      color="income"
                    >
                      +$2,354.00
                    </span>
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L17.81 9.94l-3.75-3.75L3.15 17.1c-.1.1-.15.22-.15.36zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </button>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-npdkjk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteForeverRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="css-1mavqmw"
                  draggable="true"
                  style="background-color: inherit;"
                >
                  <p>
                    income
                     from 
                    Work
                  </p>
                  <div
                    class="css-1ub4dmn-MuiStack-root"
                  >
                    <span
                      class="css-81iyw9"
                      color="income"
                    >
                      +$836.00
                    </span>
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L17.81 9.94l-3.75-3.75L3.15 17.1c-.1.1-.15.22-.15.36zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </button>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-npdkjk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteForeverRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="css-142vajq-MuiStack-root"
              >
                <nav
                  aria-label="pagination navigation"
                  class="MuiPagination-root MuiPagination-outlined css-oqcxxb-MuiPagination-root"
                >
                  <ul
                    class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                  >
                    <li>
                      <button
                        aria-label="Go to previous page"
                        class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-tnzq9a-MuiSvgIcon-root-MuiPaginationItem-icon"
                          data-testid="NavigateBeforeIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </button>
                    </li>
                    <li>
                      <button
                        aria-current="true"
                        aria-label="page 1"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-selected MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        1
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 2"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        2
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 3"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        3
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 4"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        4
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 5"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        5
                      </button>
                    </li>
                    <li>
                      <div
                        class="MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-ellipsis css-1hfyor7-MuiPaginationItem-root"
                      >
                        …
                      </div>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 10000"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        10000
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to next page"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeSmall MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-previousNext css-evpp5f-MuiButtonBase-root-MuiPaginationItem-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-tnzq9a-MuiSvgIcon-root-MuiPaginationItem-icon"
                          data-testid="NavigateNextIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                          />
                        </svg>
                      </button>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
